includes/tracks/class-wc-tracks.php

```diff
@@ -0,0 +1,155 @@
+<?php
+/**
+ * PHP Tracks Client
+ *
+ * @class   WC_Tracks
+ * @package WooCommerce/Classes
+ *
+ * Example Usage:
+ *
+```php
+	require_once( dirname( __FILE__ ) . '/libraries/tracks/class-wc-tracks.php' );
+	$result = WC_Tracks::record_event( 'wca_test_update_product', array() );
+
+	if ( is_wp_error( $result ) ) {
+		// Handle the error in your app
+	}
+```
+ */
+
+/**
+ * Class WC_Tracks
+ */
+class WC_Tracks {
+	// @TODO: Find a good prefix.
+	const PREFIX = 'wca_test_';
+	/**
+	 * Get the identity to send to tracks.
+	 *
+	 * @TODO: Determine the best way to identify sites/users with/without Jetpack connection.
+	 *
+	 * @param int $user_id User id.
+	 * @return array Identity properties.
+	 */
+	public static function get_identity( $user_id ) {
+		$has_jetpack = class_exists( 'Jetpack' );
+
+		// Meta is set, and user is still connected.  Use WPCOM ID.
+		$wpcom_id = $has_jetpack && get_user_meta( $user_id, 'jetpack_tracks_wpcom_id', true );
+		if ( $wpcom_id && Jetpack::is_user_connected( $user_id ) ) {
+			return array(
+				'_ut' => 'wpcom:user_id',
+				'_ui' => $wpcom_id,
+			);
+		}
+
+		// User is connected, but no meta is set yet.  Use WPCOM ID and set meta.
+		if ( $has_jetpack && Jetpack::is_user_connected( $user_id ) ) {
+			$wpcom_user_data = Jetpack::get_connected_user_data( $user_id );
+			add_user_meta( $user_id, 'jetpack_tracks_wpcom_id', $wpcom_user_data['ID'], true );
+
+			return array(
+				'_ut' => 'wpcom:user_id',
+				'_ui' => $wpcom_user_data['ID'],
+			);
+		}
+
+		// User isn't linked at all.  Fall back to anonymous ID.
+		$anon_id = get_user_meta( $user_id, 'jetpack_tracks_anon_id', true );
+		if ( ! $anon_id ) {
+			$anon_id = WC_Tracks_Client::get_anon_id();
+			add_user_meta( $user_id, 'jetpack_tracks_anon_id', $anon_id, false );
```

Comment:
Without unique (the false 4th prop) wouldn't this make multiple meta entries with the same key?

Reply:
In https://github.com/woocommerce/woocommerce/pull/22676 I change to `update_user_meta` but I'll make a note to ensure multiple entries can't happen.